csrc/ops/composite.cpp

```diff
@@ -304,6 +304,10 @@ static TensorView* newForMatmul(TensorView* tv_a, TensorView* tv_b) {
     out_domain[idx] = ops::newOutputIterDomain(input_ids);
   }
 
+  out_domain[ndims_out - 1] = IterDomainBuilder(mapping_a.back())
```

Comment:
I placed the rK dimension last in the output, just because that makes it a bit easier to handle the many cases we can encounter for `matmul`. Note that this does not need to match the position of the K dimension in a translated `MmaOp`.

Reply:
Should we use both, mapping_a and mapping_b here?

Reply:
Good question. I was assuming the K dimension would be `IterType::Iteration` in both operands. They will be exact mapped, so they should be equivalent to one another so I just took the ID from A.

Reply:
What happens if one is `Symbolic` and the other `Concrete` tensor?

Reply:
Good point. I updated `ops::newOutputIterDomain` to have a `std::optional<IterType> force_iter_type` argument that I am now using to create the reduction domain. That way, if the A ID is Symbolic but the B ID is not, then we'll use the B extent.